src/DataStructures/Tensor/EagerMath/Magnitude.hpp

```diff
@@ -48,11 +50,51 @@ Scalar<DataType> magnitude(
 /// \details
 /// Computes the square root of the absolute value of the scalar.
 template <typename DataType>
-Scalar<DataType> sqrt_magnitude(
-    const Scalar<DataType>& input) noexcept {
+Scalar<DataType> sqrt_magnitude(const Scalar<DataType>& input) noexcept {
   return Scalar<DataType>{sqrt(abs(get(input)))};
 }
 
+namespace L2Norm_detail {
+template <typename DataType, typename Symm, typename IndexList>
+Scalar<DataType> pointwise_l2_norm_square(
+    const Tensor<DataType, Symm, IndexList>& tensor) noexcept {
+  auto pointwise_l2_normsq = make_with_value<DataType>(tensor, 0.);
+  for (auto it = tensor.begin(); it != tensor.end(); ++it) {
+    // In order to handle tensory symmetries, we multiply the square of each
+    // stored component with its multiplicity
+    pointwise_l2_normsq += tensor.multiplicity(it) * square(*it);
+  }
+  return Scalar<DataType>{pointwise_l2_normsq};
+}
+}  // namespace L2Norm_detail
+
+/// \ingroup TensorGroup
+/// \brief Compute point-wise $\\ell^2$ (Frobenius) norm of a tensor
+///
+/// \details
+/// Computes the point-wise Frobenius norm of a given tensor of arbitrary rank.
+template <typename DataType, typename Symm, typename IndexList>
+Scalar<DataType> pointwise_l2_norm(
+    const Tensor<DataType, Symm, IndexList>& tensor) noexcept {
+  return Scalar<DataType>{
+      sqrt(get(L2Norm_detail::pointwise_l2_norm_square(tensor)))};
+}
+
+/// \ingroup TensorGroup
+/// \brief Compute $\\ell^2$ (Frobenius) norm of a tensor
+///
+/// \details
+/// Computes the Frobenius norm of a given tensor of arbitrary rank,
+/// averaged over all grid points.
```

Comment:
Also give a latex expression here. And move the `averaged over all grid points` to the `\brief`.